cmd/brimstone/main.go

```diff
@@ -20,87 +20,79 @@ import (
 	"github.com/labstack/echo/v4/middleware"
 
 	"gorm.io/driver/postgres"
+	"gorm.io/driver/sqlite"
 	"gorm.io/gorm"
+
+	"github.com/caarlos0/env/v6"
 )
 
 const (
 	GG_HEADER = "Gitguardian-Signature"
 )
 
 var (
-	version                  = "dev"
-	TLS_SKIP_VERIFY          = false
-	DEBUG                    = false
-	API_KEY_VARNAME          = "BRIMSTONE_API_KEY"
-	GG_API_TOKEN_VARNAME     = "GG_API_TOKEN"
-	GG_WEBHOOK_TOKEN_VARNAME = "GG_WEBHOOK_TOKEN"
-
-	// postgres://<username>:<password>@<host>:<port>/<database>?<parameters>
-	LOCAL_DATABASE_URL = "postgresql://root@localhost:26257/brimstone?sslmode=disable&application_name=brimstone"
+	version = "dev"
+	DEBUG   = false
 )
 
-func main() {
-	url := flag.String("hmslurl", "https://api.hasmysecretleaked.com", "HMSL url where to send hashes (Used as audience when sending JWT request)")
-	audiencetype := flag.String("hmslaudtype", "hmsl", "Audience type for HMSL JWT request")
-	ggapiurl := flag.String("ggapiurl", "https://api.gitguardian.com", "GG API URL")
-	ggapitokenvar := flag.String("ggapitokenvar", GG_API_TOKEN_VARNAME, "GG API Token env var contains GG API token to use (default: GG_API_TOKEN)")
-	ggwebhooktokenvar := flag.String("ggwebhooktokenvar", GG_WEBHOOK_TOKEN_VARNAME, "GG API Token env var contains GG custom webhook token from settings (default: GG_WEBHOOK_TOKEN)")
-
-	apikeyvarname := flag.String("keyvar", API_KEY_VARNAME, "Name of env var from which to set brimstone's api key (default: BRIMSTONE_API_KEY)")
-
-	dburl := flag.String("dburl", LOCAL_DATABASE_URL, "Database URL")
-	port := flag.String("port", "9191", "port whereon brimstone listens (default: 9191)")
+type config struct {
+	HmslUrl        string `env:"HMSL_URL" envDefault:"https://api.hasmysecretleaked.com"`
+	AudienceType   string `env:"HMSL_AUDIENCE_TYPE" envDefault:"hmsl"`
+	GgApiUrl       string `env:"GG_API_URL" envDefault:"https://api.gitguardian.com"`
+	GgApiToken     string `env:"GG_API_TOKEN,unset"`
+	GgWebhookToken string `env:"GG_WEBHOOK_TOKEN,required,unset"`
+	ApiKey         string `env:"BRIMSTONE_API_KEY,unset"`
+
+	DbUrl string `env:"DB_URL,required,unset"`
+	Port  uint16 `env:"PORT" envDefault:"9191"`
+
+	IdTenantUrl string `env:"ID_TENANT_URL,required"`
+	PcloudUrl   string `env:"PCLOUD_URL,required"`
+	SafeName    string `env:"SAFE_NAME,required"`
+	PlatformID  string `env:"PLATFORM_ID,required"`
+	PamUser     string `env:"PAM_USER,required"`
+	PamPass     string `env:"PAM_PASS,required,unset"`
+
+	TlsSkipVerify bool `env:"TLS_SKIP_VERIFY" envDefault:"false"`
+}
 
-	idtenanturl := flag.String("idtenanturl", "", "PAM config ID tenant URL")
-	pcloudurl := flag.String("pcloudurl", "", "PAM config Privilege Cloud URL")
-	pamuser := flag.String("pamuser", "", "PAM config PAM User")
-	pampass := flag.String("pampass", "", "PAM config PAM Pass")
-	safename := flag.String("safename", "", "PAM config PAM Safe Name")
+func main() {
+	e := echo.New()
```

Comment:
Why was this moved to here?  Prefer to keep this initialization closer to where it is being used.

Reply:
`e` is used below to log the error if loading the env variables fails.